### PR TITLE
Improved frame plugin testing

### PR DESCRIPTION
**Pull request by jessemapel**
This is not exhaustive but it's a start.

### CODE REVIEW

File: tests/TestyMcTestFace.cpp [hunk 1]
```diff
@@ -78,12 +91,48 @@ TEST(FramePluginTests, MissingStateValue) {
...
+   UsgsAstroFramePlugin testPlugin;
+   csm::Model *cameraModel;
+   EXPECT_NO_THROW(
+         cameraModel = testPlugin.constructModelFromISD(
+               isd,
+               "USGS_ASTRO_FRAME_SENSOR_MODEL",
+               NULL)
+   );
+   UsgsAstroFrameSensorModel *frameModel = dynamic_cast<UsgsAstroFrameSensorModel *>(cameraModel);
+   EXPECT_FALSE(frameModel == NULL);
```
Comment (jessemapel) [inline tests/TestyMcTestFace.cpp]: There's gotta be a better way to check if a pointer is NULL in gtest, but this is the best I could find.

File: tests/TestyMcTestFace.cpp [hunk 2]
```diff
@@ -78,12 +91,45 @@ TEST(FramePluginTests, MissingStateValue) {
...
+   UsgsAstroFramePlugin testPlugin;
+   // Remove the model_name keyword from the ISD to make it invalid
+   isd.clearParams("model_name");
+   csm::Model *cameraModel;
+   EXPECT_THROW(
+         testPlugin.constructModelFromISD(
+               isd,
+               "USGS_ASTRO_FRAME_SENSOR_MODEL",
+               NULL),
+         csm::Error
```
Comment (jessemapel) [inline tests/TestyMcTestFace.cpp]: This is currently just checking the we caught a generic CSM error. I'd also like to check that it's the correct error type and the message is correct but I don't know how to do that in gtest

File: tests/TestyMcTestFace.cpp (hunk 1 above)
Comment (jlaura) [reply tests/TestyMcTestFace.cpp]: What about `IsNull`? https://github.com/google/googletest/blob/master/googlemock/docs/CheatSheet.md#generic-comparison

File: tests/TestyMcTestFace.cpp [hunk 3]
```diff
@@ -18,8 +20,19 @@ class FrameIsdTest : public ::testing::Test {
       json jsonIsd = json::parse(isdFile);
       isd.clearAllParams();
```
Comment (jlaura) [inline tests/TestyMcTestFace.cpp]: Is the setUp rerun each time and it is necessary to clear the ISD before a second run?

File: tests/TestyMcTestFace.cpp [hunk 4]
```diff
@@ -18,8 +20,19 @@ class FrameIsdTest : public ::testing::Test {
...
       for (json::iterator it = jsonIsd.begin(); it != jsonIsd.end(); ++it) {
-         isd.addParam(it.key(), it.value().dump());
+         if (it.value().size() >1 ) {
+            std::vector<double> v = it.value();
+            for (int j=0;j < v.size(); j++) {
```
Comment (jlaura) [inline tests/TestyMcTestFace.cpp]: I thought in the JSON library it is not necessary to cast this into a standard vector? https://github.com/nlohmann/json#stl-like-access

File: tests/TestyMcTestFace.cpp [hunk 5]
```diff
@@ -18,8 +20,19 @@ class FrameIsdTest : public ::testing::Test {
...
+               std::ostringstream val;
+               val << std::setprecision(15) << v[j];
+               isd.addParam(it.key(),val.str());
```
Comment (jlaura) [inline tests/TestyMcTestFace.cpp]: I would rather see precision being set in a single place - inside of the ISD.

File: tests/TestyMcTestFace.cpp (hunk 2 above)
Comment (jlaura) [reply tests/TestyMcTestFace.cpp]: https://stackoverflow.com/questions/23270078/test-a-specific-exception-type-is-thrown-and-the-exception-has-the-right-propert

File: tests/TestyMcTestFace.cpp (hunk 1 above)
Comment (jessemapel) [reply tests/TestyMcTestFace.cpp]: :+1:

File: tests/TestyMcTestFace.cpp (hunk 2 above)
Comment (jessemapel) [reply tests/TestyMcTestFace.cpp]: I was hoping for a better way to do that. I feel like there should be an approach that doesn't use the fail assertion.

**Comment by jlaura** [comment]
I only caught the most recent change and approved - other comments are still inline.

File: tests/TestyMcTestFace.cpp (hunk 3 above)
Comment (jessemapel) [reply tests/TestyMcTestFace.cpp]: setUp is called before every test that uses the framework. We actually don't need to clear the ISD

File: tests/TestyMcTestFace.cpp (hunk 5 above)
Comment (jessemapel) [reply tests/TestyMcTestFace.cpp]: I think the precision can just be removed

**Comment by jlaura** [comment]
@jessemapel This looks good to me. Are you happy with a merge?

**Pull request merged**